recipes/libsamplerate/all/conanfile.py

```diff
@@ -74,12 +74,18 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "SampleRate")
+        self.cpp_info.set_property("cmake_target_name", "SampleRate::samplerate")
+        self.cpp_info.set_property("pkg_config_name", "samplerate")
+        # TODO: back to global scope once cmake_find_package* generators removed
+        self.cpp_info.components["samplerate"].libs = ["samplerate"]
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.components["samplerate"].system_libs.append("m")
+
+        # TODO: to remove in conan v2 once cmake_find_package* & pkg_config generators removed
         self.cpp_info.names["cmake_find_package"] = "SampleRate"
         self.cpp_info.names["cmake_find_package_multi"] = "SampleRate"
         self.cpp_info.names["pkg_config"] = "samplerate"
         self.cpp_info.components["samplerate"].names["cmake_find_package"] = "samplerate"
         self.cpp_info.components["samplerate"].names["cmake_find_package_multi"] = "samplerate"
```

Comment:
will the global package and the samplerate both generate samplerate.pc ?

Reply:
yes

Reply:
it seems brittle to me. is it documented that global level files are generated before component level files ?

Reply:
no, and actually `PkgConfigDeps` is not robust to this trick, it creates a file including itself /cc @czoido

Reply:
why not simply using another name for the global one ?

Reply:
to avoid creating an unofficial pkg-config file. Once we start to create targets or pkg-config files, people start using them, then complain because some new revision removed this file/target which was not supposed to be there. It's like exposing private headers.

Reply:
then we can name it something like "samplerateThisFileWillBeRemovedSoonDoNotUseIt"

Reply:
yes, sounds like a relying on the implementation details, which is too fragile.

Reply:
I have commented here: https://github.com/conan-io/conan-center-index/pull/8623/files#r783285615
In `PkgConfigDeps` this should not be a problem while we transition to 2.0 if we don't set the `pkg_config_name` name for the component, the generator will create a file `samplerate.pc` for the root cpp_info that includes a `samplerate-samplerate.pc` from the component.
The `pkg_config` generator will overwrite those files as it does not add the `-` to the component filename, maybe at least in this case we can change the name of the component as the only reason to be there is modifying the namespace of the CMake target.

Reply:
I was discussing this morning with the team if we now a legit case where there has to be a component with the same name as the name of the root cpp_info (at least for `pkg-config`) If there's not such case we could add a hook to check this condition so that we don't have to rely on that fragile "feature" of the `pkg_config` generator overwritting the file of the root cpp_info. Also, for the component keys probably we should always add the `_` character or something to avoid this same problem for components that do not set the name.
cc @ericLemanissier @SpaceIm @SSE4 @jgsogo @uilianries

Reply:
@SpaceIm I agree that poluting with this files is ugly, but this is somehow transparent for the user, and something that would only happen until Conan 2.0. After 2.0 these components will disappear and we will not have this problem any more unless there is some library that specifically needs to define a component with the same name as the root cpp_info...

Reply:
I'm just a little bit tired to see issues from users who were relying on some "unofficial" target name / pkg config files, then complaining that a change in recipe interface broke their CI / local build. I understand why they complain, in a sense they just trust conan-center to provide good targets / pkg files from scratch, and use whatever is generated.

Reply:
Thanks for the feedback @SpaceIm Taking this into account, probably then the best approach would be testing and documenting that the `.pc` files are generated in a certain order and then enabling the use of this feature that would be no longer accidental or fragile and making. WDYT?

Reply:
I agree